hera_qm/utils.py

```diff
@@ -442,3 +442,28 @@ def metrics2mc(filename, ftype):
         raise ValueError('Metric file type ' + ftype + ' is not recognized.')
 
     return d
+
+
+def dynamic_slice(arr, slice_obj, axis=-1):
+    """
+    Dynamic slicing of arr along axis given slice object.
+
+    Parameters
+    ----------
+    arr : ndarray
+
+    slice_obj : slice object
+
+    axis : integer
+
+    Returns
+    -------
+    arr_slice : ndarray
+    """
+    if isinstance(axis, (int, np.integer)):
+        axis = (axis,)
+    assert isinstance(arr, np.ndarray), "arr must be an ndarray"
```

Comment:
Could you put this in an if statement instead? Paul has pointed out you can run code in a way that ignore asserts, so we're generally avoiding them in hera_qm.